### PR TITLE
Najla-week3-js

### PR DESCRIPTION
**Pull request by NajlaSoltana**

### CODE REVIEW

File: week-3/A-array-find/exercise.js [hunk 1]
```diff
@@ -6,9 +6,14 @@
 // write your code here
 
 var names = ["Rakesh", "Alexandra", "Annam", "Mikey", "Karim", "Ahmed"];
-var longNameThatStartsWithA = findLongNameThatStartsWithA(names);
+var longNameThatStartsWithA = names.find(findLongNameThatStartsWithA);
+
+function findLongNameThatStartsWithA(names) {
+  return names.length >= 7 && names[0] === "A";
```
Comment (martynamaron) [inline week-3/A-array-find/exercise.js]: Great stuff! Although the instructions say "longer than" 7 - but that's a minor point :)

File: week-3/B-array-some/exercise.js [hunk 2]
```diff
@@ -15,10 +15,11 @@ var pairsByIndex = [[0, 3], [1, 2], [2, 1], null, [3, 0]];
...
 var mentors = ["Daniel", "Irina", "Mozafar", "Luke"];
 
-var pairs = pairsByIndex.map(function(indexes) {
-  var student = students[indexes[0]];
-  var mentor = mentors[indexes[1]];
-  return [student, mentor];
+var containNull = pairsByIndex.some(function (include) {
+  include === null;
+  console.log(" null value is found in this array");
+  process.exit(1)
```
Comment (martynamaron) [inline week-3/B-array-some/exercise.js]: So the purpose of this exercise was to **check** if the array contains a `null` value. If it does - call `process.exit(1)`. You can **check** something with a conditional - think about what the `some()` method returns and how you can use this.

Also, the instructions say - don't edit any of the existing code - just add your own check underneath `pairsByIndex` declaration.

Try again! :)

File: week-3/C-array-every/exercise.js [hunk 3]
```diff
@@ -4,15 +4,24 @@
...
+      console.log("The group contains only students");
+    } else {
+      console.log("The group does not contain only students");
+    }
 }
 
 /* EXPECTED RESULT */
 
 // The group does not contain only studnets
+// well done Najla, expected result met
```
Comment (martynamaron) [inline week-3/C-array-every/exercise.js]: Is this a note to yourself? :D

File: week-3/D-array-filter/exercise.js [hunk 4]
```diff
@@ -8,15 +8,23 @@
 
 var pairsByIndexRaw = [[0, 3], [1, 2], [2, 1], null, [1], false, "whoops"];
 
-var pairsByIndex; // Complete this statement
+var pairsByIndex = pairsByIndexRaw.filter(arrayElemnet => {
```
Comment (martynamaron) [inline week-3/D-array-filter/exercise.js]: Woooop! 🎉 Well done! And a very simple and readable solution here! :)

File: week-3/E-array-map/exercise.js [hunk 5]
```diff
@@ -3,3 +3,9 @@
 
 var numbers = [0.1, 0.2, 0.3, 0.4, 0.5];
 
+var multiplyNumbers = numbers.map(number => {
+    return number * 100;
```
Comment (martynamaron) [inline week-3/E-array-map/exercise.js]: Very well! 
The instructions also mention writing multiple solutions using different syntax - could you add those examples too?

File: week-3/F-array-forEach/exercise.js [hunk 6]
```diff
@@ -8,6 +8,17 @@
 */
 
 var arr = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15];
+arr.forEach(function (number) {
```
Comment (martynamaron) [inline week-3/F-array-forEach/exercise.js]: Great!
Just a minor point - the expected output omits the `0` - so there should be no printout for it. At the moment though, it satisfies the `number % 3 === 0 && number % 5 === 0` condition. How can we exclude `0`?

**Comment by martynamaron** [comment]
Great stuff @NajlaSoltana ! Well done raising the PR with just the week-3 exercises 👏 
I've left a couple comments on `some()` and `every()` - please try again with those :) Thanks!

File: week-3/B-array-some/exercise.js (hunk 2 above)
Comment (NajlaSoltana) [reply week-3/B-array-some/exercise.js]: I'll do.

File: week-3/C-array-every/exercise.js (hunk 3 above)
Comment (NajlaSoltana) [reply week-3/C-array-every/exercise.js]: yes it is. I need some encouragement :D

File: week-3/C-array-every/exercise.js [hunk 7]
```diff
@@ -4,15 +4,24 @@
...
-var groupIsOnlyStudents; // complete this statement
+var groupIsOnlyStudents = group.every(checkForStudents); // complete this statement
 
 if (groupIsOnlyStudents) {
   console.log("The group contains only students");
 } else {
-  console.log("The group does not contain only students");
+  console.log("The group does not contain only students");}
+
+  function checkForStudents(groupIsOnlyStudents) {
```
[the remark this answers is not in the document]
Comment (NajlaSoltana) [reply week-3/C-array-every/exercise.js]: OK. I'll do

File: week-3/C-array-every/exercise.js (hunk 3 above)
Comment (martynamaron) [reply week-3/C-array-every/exercise.js]: That's cool! Positive thinking, I like it :)

File: week-3/E-array-map/exercise.js (hunk 5 above)
Comment (NajlaSoltana) [reply week-3/E-array-map/exercise.js]: I'll do.

**Pull request closed without merging**